test/rest/RestClientTests.cs

```diff
@@ -1,348 +1,368 @@
+
+            // Act
+            RestResponse actualResponse = restClient.Put(putResourceContext);
+
+            // Assert
+            Assert.IsNotNull(actualResponse);
+            Assert.AreEqual(expectedResponse.StatusCode, actualResponse.StatusCode);            
+        }
+
+        [Test]
+        public void Put_HttpReturnsNull_ReturnsNull()
+        {
+            // Arrange
+            string content = "content";
+
+            PutRestResourceContext putResourceContext = new PutRestResourceContext(content);
+
+            HttpRequest httpRequest = new HttpRequest()
+            {
+                HttpMethod = HttpMethod.PUT,
+                ResourcePath = putResourceContext.ResourcePath,
+                Content = putResourceContext.Content,
+                ContentType = "application/xml",
+                ContentEncoding = Encoding.UTF8
+            };
+
+            RestResponse expectedResponse = new RestResponse()
+            {
+                StatusCode = HttpStatusCode.NotFound
+            };
+
+            HttpResponse httpResponse = null;
+
+            mockHttpClient.Setup(hc => hc.Submit(httpRequest)).Returns(httpResponse);
+
+            // Act
+            RestResponse actualResponse = restClient.Put(putResourceContext);
+
+            // Assert
+            Assert.IsNull(actualResponse);
+        }
+
+        [Test]
+        public void Delete_HttpReturnsIn200Range_ReturnsRestResponse()
+        {
+            // Arrange
+            DeleteRestResourceContext deleteResourceContext = new DeleteRestResourceContext();
+
+            HttpRequest httpRequest = new HttpRequest()
+            {
+                HttpMethod = HttpMethod.DELETE,
+                ResourcePath = deleteResourceContext.ResourcePath,
+                ContentType = "text/plain",
+            };
+
+            RestResponse expectedResponse = new RestResponse()
+            {
+                StatusCode = HttpStatusCode.OK
+            };
+
+            HttpResponse httpResponse = new HttpResponse()
+            {
+                StatusCode = HttpStatusCode.OK
+            };
+
+            mockHttpClient.Setup(hc => hc.Submit(httpRequest)).Returns(httpResponse);
+
+            // Act
+            RestResponse actualResponse = restClient.Delete(deleteResourceContext);
+
+            // Assert
+            Assert.IsNotNull(actualResponse);
+            Assert.AreEqual(expectedResponse.StatusCode, actualResponse.StatusCode);
+        }
+
+        [Test]
+        public void Delete_HttpReturnsNull_ReturnsNull()
+        {
+            // Arrange
+            DeleteRestResourceContext deleteResourceContext = new DeleteRestResourceContext();
+
+            HttpRequest httpRequest = new HttpRequest()
+            {
+                HttpMethod = HttpMethod.DELETE,
+                ResourcePath = deleteResourceContext.ResourcePath,
+                ContentType = "text/plain"
+            };
+
+            RestResponse expectedResponse = new RestResponse()

```

Comment:
Unused - but this isn't new? I wonder if we can do a ReSharper solution format and remove these redundancies...?